fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java

```diff
@@ -2213,17 +2231,18 @@ Bundle createSearchResponseBundle(List<Resource> resources, FHIRSearchContext se
 
             FHIRPathEvaluator evaluator = FHIRPathEvaluator.evaluator();
 
-
             // Loop through the resources, looking for versioned references and references to multiple resource types for the same logical ID
-            for (Resource resource : matchResources) {
+            for (ResourceResult<? extends Resource> resourceResult : matchResources) {
+                Resource resource = resourceResult.getResource();
+
                 // A flag that indicates whether we need to take a closer look at the reference values or not
                 boolean needsEval = false;
 
                 // If any of the reference values are "versioned"
                 // then we'll need to check that they aren't used for chaining
                 // TODO Should we pass the previously-gathered set of references into the method instead?
                 CollectingVisitor<Reference> refCollector = new CollectingVisitor<>(Reference.class);
-                resource.accept(refCollector);
+                resource.accept(refCollector); // TODO protect against null resource?
```

Comment:
how to resolve?

Reply:
Added a check before we reach this. As it is unexpected, we log and throw an exception.